projects/wave-core-new/src/lib/operators/dialogs/helpers/layer-selection/layer-selection.component.ts

```diff
@@ -85,18 +86,23 @@ export class LayerSelectionComponent implements OnChanges, OnDestroy, ControlVal
         if (changes.layers || changes.types) {
             if (this.layers instanceof Observable) {
                 this.layers.pipe(first()).subscribe(layers => {
-                    this.filteredLayers.next(
-                        layers.filter((layer: Layer) => {
-                            return this.types.map(t => t.getCode()).indexOf(layer.layerType) >= 0;
+                    let o = layers.map(l => this.projectService.getLayerMetadata(l));
```

Comment:
1. You could give `o` a name.
2. Why don't you use pipe and mergeMap instead of double-subscribe?

Reply:
I'm not that familiar with rxjs yet, maybe you can help me. Instead of subscribing I move it into a mergeMap inside the pipe call or how do I do it?

Reply:
Yeah, exactly.
for `mergeMap`, you get the layer as input (like in map) and you return the forkJoin.
It is like flatMap but changes its names with every other RxJs version.
This way, you only need to subscribe once.
If you have any trouble, we can take a look together.

Reply:
8821af3fa27455584d4627494cb03f9cd4d3270c